cni-plugin/pkg/install/install.go

```diff
@@ -496,3 +504,46 @@ func setSuidBit(file string) error {
 
 	return nil
 }
+
+func createCNIToken(kubecfg *rest.Config) string {
+	clientset, err := kubernetes.NewForConfig(kubecfg)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to create clientset for CNI kubeconfig")
+	}
+
+	supported := false
+	once := &sync.Once{}
+	tokenRequestSupported := func() bool {
+		once.Do(func() {
+			resources, err := clientset.Discovery().ServerResourcesForGroupVersion("v1")
+			if err != nil {
+				return
+			}
+			for _, resource := range resources.APIResources {
+				if resource.Name == "serviceaccounts/token" {
+					supported = true
+					return
+				}
+			}
+		})
+		return supported
+	}
+
+	validity := int64(defaultCNITokenValiditySeconds)
+	tr := &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			Audiences:         []string{"kubernetes"},
+			ExpirationSeconds: &validity,
+		},
+	}
+
+	tokenRequest, err := clientset.CoreV1().ServiceAccounts(metav1.NamespaceSystem).CreateToken(context.TODO(), "calico-node", tr, metav1.CreateOptions{})
```

Comment:
Do we need to garbage collect old tokens after they have expired? Or will that be done automatically by k8s?

Reply:
I will check this.
In my initial tests I did not seem to be able to get any of the created tokens after creation. Therefore I would not know how to garbage collect them, but I will take a deeper look.

Reply:
Looking how `kubelet` manages the tokens, it does not look like garbage collection is required (see https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/token/token_manager.go). Tokens are created and cached in `kubelet`, but only the local cache is cleaned up. There is no interaction with ´kube-apiserver` to delete tokens.
I will check with `kube-apiserver` separately, though.

Reply:
There is no API to get issued tokens or delete them. Hence, there is also no way to garbage collect them.
My understanding is that tokens will simply expire and then no longer be useful.

@caseydavenport Is that sufficient or did you have something else in mind?